test/integration/bidirectional-full-sync-test.js

```diff
@@ -20,48 +20,114 @@
+        nodeB.dissemination.clearChanges();
+
+        // verify nodeA does include B in it's member list
+        assert.notEqual(nodeA.membership.findMemberByAddress(nodeB.hostPort), undefined);
+
+        // verify nodeB doesn't include A in it's member list
+        assert.equal(nodeB.membership.findMemberByAddress(nodeA.hostPort), undefined);
+
+        nodeB.membership.on('updated', function(updates) {
+            // verify nodeB now includes A in it's member list
+            assert.notEqual(nodeB.membership.findMemberByAddress(nodeA.hostPort), undefined);
+            assert.end();
+        });
+
+        nodeA.gossip.tick();
     });
+
+    assert.timeoutAfter(2000);
 });
 
-function mkBadPingResponder(ringpop) {
-    ringpop.channel.register('/protocol/ping', function protocolPing(req, res) {
-        res.headers.as = 'raw';
-        res.sendNotOk(null, JSON.stringify('ping failed on purpose'));
-    });
-}
 
 testRingpopCluster({
+    bootstrapSize: 1,
     size: 2,
     waitForConvergence: false,
-    autoGossip: false,
-    tap: function tap(cluster) {
-        mkBadPingResponder(cluster[1]);
-    },
-}, 'don\'t raise piggyback counter when ping fails', function t(bootRes, cluster, assert) {
-    assert.plan(1);
-
-    var pingSender = cluster[0];
-    pingSender.gossip.tick(function() {
-        var changes = pingSender.dissemination.changes;
-        var joinChange = changes[pingSender.whoami()];
-        var piggybackCount = joinChange.piggybackCount;
-        assert.equals(piggybackCount, 0, 'piggyback counter not raised');
-        assert.end();
+    autoGossip: false
+}, 'test bi-directional full syncs throttles', function t(bootRes, cluster, assert) {
+    // nodeA will include B in it's member list
+    var nodeA = _.find(cluster, function(node) { return node.isReady;});
+
+    // nodeB will not include A in it's member list
+    var nodeB = _.find(cluster, function(node) { return !node.isReady;});
+
+    // bootstrap node B without A.
+    nodeB.bootstrap([nodeB.hostPort], function onBootStrapped(err) {
+        assert.error(err);
+
+        // verify nodeA does include B in it's member list
+        assert.notEqual(nodeA.membership.findMemberByAddress(nodeB.hostPort), undefined);
+
+        // verify nodeB doesn't include A in it's member list
+        assert.equal(nodeB.membership.findMemberByAddress(nodeA.hostPort), undefined);
+
+        var realProtocolJoin = nodeB.client.protocolJoin.bind(nodeB.client);
+        var protocolJoinCalls = [];
+        nodeB.client.protocolJoin = function() {
+            protocolJoinCalls.push(arguments);
+            assert.equal(nodeB.dissemination.reverseFullSyncJobs, protocolJoinCalls.length);
+            if (protocolJoinCalls.length > nodeB.maxReverseFullSyncJobs) {
+                assert.fail('full sync should have been throttled');
+            }
+        };
+
+        async.timesSeries(nodeB.maxReverseFullSyncJobs, function tick(i, next) {
```

Comment:
Timeseries makes it easier to do something x-times in series instead of in parallel. Because I'm calling `next()` in the callback I'm sure that the joins are all complete in series instead of in parallel.